### ISSUE TITLE
Enable 8 bitdepth support

### ISSUE THREAD

**Issue by randomPoison** [memorysafety/rav1d#3]
The [initial transpile](https://github.com/memorysafety/rav1d/issues/1) of dav1d only enabled 16 bitdepth, so we need to re-enable support for 8 bitdepth.

The issue is the the `*_tmpl.c` files in dav1d get compiled multiple times with the `BITDEPTH` define set to different values (either 8 or 16), and c2rust can't currently handle transpiling the same source file multiple times. We'll likely need to manually transpile the project for 8 bitdepth and integrate the alternate version of the `_tmpl.rs` files that are generated. See [this comment](https://github.com/memorysafety/rav1d/issues/1#issuecomment-1439294472) for the steps that were used to do the initial transpile. In the `meson configure` step, change `"-Dbitdepths=['16']"` to `"-Dbitdepths=['8']"` to configure the build for 8 bitdepth.

We also need to figure out how to make the different bitdepth options configurable as build options for Rust. Maybe setup features for `8_bitdepth` and `16_bitdepth` and have both features enabled by default?

**Comment by randomPoison** [comment]
As of 8a51f0d support for 8 bitdepth has been transpiled from dav1d, and there are now two Cargo features `bitdepth_8` and `bitdepth_16` to control which is enabled. The 8 bitdepth data tests seem to all pass, but I'm noticing that a handful of tests in that suite timeout when run. I need to investigate further to determine what's going on there.

**Comment by randomPoison** [comment]
Turns out the tests were timing out because I was running tests against debug builds. For some reason the Rust debug build runs substantially slower than the C debug build, so when the full test suite is running with multiple tests running in parallel some of the tests can take more than 30 seconds to complete. Running the tests against a release build allows the full suite to complete successfully.

I also did a cursory comparison of how long tests take to run for the C version (with asm disabled) vs the Rust version, and in release mode both seem to take a comparable amount of time. So that's a good indication that there's no major performance regression going from C to Rust, at least without asm support.
